### PR TITLE
cloudront.update_distribution() support

### PR DESCRIPTION
**Pull request by pll**
[ Adds update_distribution() support for cloudfront client ](https://github.com/spulec/moto/issues/5114)

- Basic update_distribution() functionality
- Tests not overly extensive, but verifies changes to an existing config get updated in the actual config.

### CODE REVIEW

File: moto/cloudfront/urls.py [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
 from .responses import CloudFrontResponse
 
+
 response = CloudFrontResponse()
 
 url_bases = [
     r"https?://cloudfront\.amazonaws\.com",
 ]
-
+# import pytest
```
Comment (bblommers) [inline moto/cloudfront/urls.py:10]: We don't need to keep this

**Review by bblommers — CHANGES_REQUESTED** [review]
Hi @pll, thanks for the PR! Can we get a separate tests for the three error scenarios as well?

File: moto/cloudfront/urls.py (hunk 1 above)
Comment (pll) [reply moto/cloudfront/urls.py:10]: Hi @bblommers,

Sorry about those commented out lines, I missed them in my cleanup...

I'm happy to write additional tests, but are you referring to my existing case and breaking that up into 3 different tests, or writing additional tests to cover scenarios not yet covered? If the latter, what scenarios did you have in mind?

Thanks.
Paul

Comment (pll) [reply moto/cloudfront/urls.py]: Oh, Never mind, I just looked at models.py, and now I realize you're asking to cover these errors:

```
        if Id not in self.distributions or Id is None:
            raise NoSuchDistribution
        if not IfMatch:
            raise InvalidIfMatchVersion
        if not DistributionConfig:
            raise NoSuchDistribution
```

I'll write those up.

File: tests/test_cloudfront/test_cloudfront.py [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+    dist_config = dist["Distribution"]["DistributionConfig"]
+    aliases = ["alias1", "alias2"]
+    dist_config["Aliases"] = {"Quantity": len(aliases), "Items": aliases}
+
+    with pytest.raises(ClientError) as error:
+        client.update_distribution(
+            DistributionConfig=dist_config, Id=dist_id, IfMatch=dist_etag
+        )
+
+    pytest.set_trace()
```
Comment (bblommers) [inline tests/test_cloudfront/test_cloudfront.py:165]: We don't need this

**Review by bblommers — CHANGES_REQUESTED** [review]
Thanks @pll, that was exactly what I meant. Just got one minor comment, everything else looks good.
You may have to resolve a merge conflict as well, I'm afraid - I have just merged a PR with massive changes to `moto/core/models.py`

File: tests/test_cloudfront/test_cloudfront.py (hunk 2 above)
Comment (pll) [reply tests/test_cloudfront/test_cloudfront.py]: Whoops! Sorry about that.

All taken care of. And I've merged master back into my branch. Just one big conflict that wasn't auto-resolved.

Tests are running.

**Review by bblommers — APPROVED** [review]
Awesome - thank you for your work on this @pll!

**Comment by pll** [comment]
@bblommers - My pleasure. Glad I can help!

**Pull request merged**